pkg/virt-launcher/virtwrap/device/hostdevice/hostdev.go

```diff
@@ -52,9 +52,39 @@ func CreatePCIHostDevices(hostDevicesData []HostDeviceMetaData, pciAddrPool Addr
 	return createHostDevices(hostDevicesData, pciAddrPool, createPCIHostDevice)
 }
 
+func isVgpuDisplaySet(hostDevicesData []HostDeviceMetaData) bool {
+	for _, hostDeviceData := range hostDevicesData {
+		if hostDeviceData.VirtualGPUOptions != nil &&
+			hostDeviceData.VirtualGPUOptions.Display != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func isMdevRamFBConfigured(hostDevices []api.HostDevice) bool {
+	for _, mdev := range hostDevices {
+		if mdev.Display == "on" && mdev.RamFB == "on" {
+			return true
+		}
+	}
+	return false
+}
+
 func CreateMDEVHostDevices(hostDevicesData []HostDeviceMetaData, mdevAddrPool AddressPooler, enableDefaultDisplay bool) ([]api.HostDevice, error) {
 	if enableDefaultDisplay {
-		return createHostDevices(hostDevicesData, mdevAddrPool, createMDEVHostDeviceWithDisplay)
+		devices, err := createHostDevices(hostDevicesData, mdevAddrPool, createMDEVHostDeviceWithDisplay)
+		if err != nil {
+			return devices, err
+		}
+		// add a default single display option with enabled ramfb
+		// only if no vgpuDisplay option was configured.
+		if !isVgpuDisplaySet(hostDevicesData) && len(devices) > 0 {
+			devices[0].Display = "on"
+			devices[0].RamFB = "on"
+		}
+		return devices, nil
```

Comment:
Why don't we keep it in `createMDEVHostDeviceWithDisplay`?

Reply:
Because we want to enable one display ramfb. It simply acts as a default.
`createMDEVHostDeviceWithDisplay` is not aware of whether it has already been configured.

Reply:
For whatever reason I thought `createMDEVHostDeviceWithDisplay` is public and could be miss-used